include/host/wasi_crypto/asymmetric_common/func.h

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+
+#include "host/wasi_crypto/ctx.h"
+#include "runtime/hostfunc.h"
+#include "runtime/instance/memory.h"
+#include "wasi_crypto/api.hpp"
+
+namespace WasmEdge {
+namespace Host {
+namespace WASICrypto {
+namespace AsymmetricCommon {
+
+template <typename T> class HostFunction : public Runtime::HostFunction<T> {
+public:
+  HostFunction(WasiCryptoContext &HostCtx)
+      : Runtime::HostFunction<T>(0), Ctx(HostCtx) {}
+
+protected:
+  WasiCryptoContext &Ctx;
+};
+
+class KeypairGenerate : public HostFunction<KeypairGenerate> {
+public:
+  using HostFunction::HostFunction;
+
+  Expect<uint32_t> body(Runtime::Instance::MemoryInstance *MemInst,
+                        uint32_t AlgorithmType, const_uint8_t_ptr AlgorithmPtr,
+                        __wasi_size_t AlgorithmLen, uint32_t Options,
```

Comment:
In the top level, we should use `stdint` like `int32_t`, `int64_t`, `uint32_t`, or `uint64_t`. Then, cast this type inside the function body.

E.g. `const_uint8_t_ptr` -> `uint32_t`; `__wasi_size_t` -> `uint32_t`.